### ISSUE TITLE
Submitting CLOB from JS variable

### ISSUE THREAD

**Issue by swesley** [foex-open-source/fos-execute-plsql-code#6]
Hi,

This may be an issue with how I'm utilising the plugin, but I've tried the following.

Define a page with the following in Function/Global variable declaration

```
var sw_test; // try with simple variable

const sw = {}; // try with namespace
sw.test = {
  theClob: '';

  ,setClob:  function(pClob) {

    sw.test.theClob = pClob;
  }

  ,getClob:  function(pClob) {

    return sw.test.theClob ;
  }

};  
```

Execute when page loads

```
sw_test = 'Here'; // define value with simple variable
sw.test.setClob('There'); // define another value with namespaced variable
```

Then I try 3 different types of invocation
This successfully returns a value in the PL/SQL code for apex_application.g_clob_01

- Submit CLOB From JavaScript Variable
- Submit From : sw_test

This has **no** value in the PL/SQL code for apex_application.g_clob_01

- Submit CLOB From JavaScript Variable
- Submit From : sw.test.theClob

And this successfully returns a value in the PL/SQL code for apex_application.g_clob_01
- Submit CLOB From Function defined in JS initialisation
- JS Init:
 
```
function(options) {
   options.submitClob = function() { return sw.test.getClob(); }
   return options;
}
```

Is there anything wrong with my use in the second scenario?

Cheers

**Comment by foex-open-source** [comment]
Hi,

Thanks for reporting the issue.

There's nothing really wrong with your use in the second scenario per se, it's just "not compatible" with the structure of the plug-in. When the "_Submit CLOB_" attribute is set to "_From Javascript Variable_", we try to access the provided variable on the **window** object, but global **const** variables are not part of the window object. From MDN: 
["...Global constants do not become properties of the window object, unlike var variables...](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const)"
The help text will be updated.

And besides that, the plug-in currently does not work with nested objects in the "_Submit CLOB_" attribute, this is a bug, it'll be fixed in the next release. Until that, your third example can serve as a great workaround!

Thanks

**Comment by foex-open-source** [comment]
Fixed with 21.2.
